### DISCUSSION TITLE
Dev server exits early in Next15 when using Turbopack

### DISCUSSION THREAD

**Discussion by MangoMarcus** [vercel/next.js#73025] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/MangoMarcus/turbopack-exit

### To Reproduce

- Install with `npm i`
- Run development server with `npm run dev`
- Open [http://localhost:3000](http://localhost:3000) in your browser
- `npm run dev` will exit with code `0`, without an error.

### Current vs. Expected behavior

`npm run dev` will exit with code `0`, without an error

### Provide environment information

```bash
Operating System:
  Platform: macOS
  Chip: Apple M1 Pro
  Version: macOS 14.4
Binaries:
  Node: 20.2.0
  npm: 9.6.6
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.3
  eslint-config-next: 15.0.3
  react: react@19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

This was not an issue before migrating to Next 15 and also only affects Turbopack, not Webpack. I've narrowed down the issue to Preline, although I'm not sure if the underlying issue is with Next, Turbopack, or Preline. As it works fine on either Next 14 or Webpack, I'm guessing it's not Preline itself that's the issue.

I've added some 'XXX' comments for context, it seems the process exits whenever a Preline import is read, eg. `console.log(typeof Preline)`

**Comment by samcx** [comment]
@MangoMarcus I'm not sure you're even supposed to import Preline outside of `<PrelineScript />` in the Layout.

I tried setting up with the documentation (they have a StackBlitz example), and it seem to work!

**Reply by MangoMarcus** [reply]
@samcx The problem isn't in importing the preline scripts (eg. `import("preline/preline")`, like in their [NextJS example](https://preline.co/docs/frameworks-nextjs.html)), it's with importing the Preline Tailwind plugin (eg. `import PrelinePlugin from 'preline/plugin'). That's what my repo shows - it only adds Preline as a Tailwind plugin, it doesn't actually use any of its components or initialise its scripts or styles.

Normally you wouldn't do this outside of the Tailwind config file, but as mentioned in my reproduction my [use case](https://github.com/MangoMarcus/turbopack-exit/blob/master/src/lib/tailwindConfig.ts) is a module where I import the Tailwind config with `import tailwindConfig from '@/../tailwind.config' in order to reuse the config in other modules - eg. screen sizes. As mentioned in the repo comments it only happens when the plugin import is read - eg. console.log(typeof PrelinePlugin) is enough to cause the dev server to exit.

Regardless of how conventional this setup is it works fine with Webpack in Next 14 and 15, or with Turbopack in Next 14, but exits without an error with Turbopack in Next 15, doesn't that constitute a bug that needs addressing? Shouldn't it never exit without an error? I'm currently unable to use Webpack due to a [different issue](https://github.com/vercel/next.js/issues/66292).

It took a long time to narrow down the bug this far so I'd appreciate it being investigated ty 🙏

**Reply by MangoMarcus** [reply]
@samcx Could you respond to this or convert back to an issue? Thanks

**Reply by samcx** [reply, marked as answer]
@MangoMarcus my apologies for the delay and prematurely moving this to discussions! I tried digging into this again, and was not able to find a culprit—my guess is it has nothing to do with Preline.

I've re-opened the issue → https://github.com/vercel/next.js/issues/76182.

**Reply by MangoMarcus** [reply]
@samcx Awesome, thanks very much!

**Comment by pavlo-js** [comment]
Yup, same Issue on me as well.

`npx create-next-app@latest` and enable Turbo pack, got same issue.

**Reply by theshadowagent** [reply]
Same issue here with next 15, Turborepo & bun

**Comment by thezzisu** [comment]
Hi guys, I've recently faced with this problem and found it MAY related to OOM issue. By increasing memory the problem seems to be solved.
